File: python/examples/existing_provider.py
from opentelemetry.sdk.trace import TracerProvider
from opentelemetry.sdk.trace.export import ConsoleSpanExporter, SimpleSpanProcessor

from confident_trace import init, shutdown, span

provider = TracerProvider()
provider.add_span_processor(SimpleSpanProcessor(ConsoleSpanExporter()))
init(tracer_provider=provider, instrumentations=())
with span("standard OTel") as current:
    current.set_attribute("example", True)
shutdown()  # only Confident's processor
provider.shutdown()  # application owns its provider

File: typescript/examples/existing-provider.ts
import { InMemorySpanExporter } from '@opentelemetry/sdk-trace-base';
import { NodeTracerProvider } from '@opentelemetry/sdk-trace-node';
import { createSpanProcessor } from '@/index';

const exporter = new InMemorySpanExporter();
const provider = new NodeTracerProvider({
  spanProcessors: [createSpanProcessor({ exporter })],
});
provider.register();
provider.getTracer('application').startSpan('work').end();
await provider.forceFlush();
console.log(exporter.getFinishedSpans().map((span) => span.name));
// The application, not confident-trace.shutdown(), owns this lifecycle.
await provider.shutdown();
